Dockerfile

```diff
@@ -1,16 +1,13 @@
-FROM alpine:3.13.5@sha256:def822f9851ca422481ec6fee59a9966f12b351c62ccb9aca841526ffaa9f748
+FROM alpine:3.14.1@sha256:be9bdc0ef8e96dbc428dc189b31e2e3b05523d96d12ed627c37aa2936653258c
 
 USER root
 
 ENTRYPOINT ["tini", "--"]
 
 RUN ["apk", "--no-cache", "upgrade"]
-RUN ["apk", "--no-cache", "add", "tini", "dnsmasq", "bash", "curl", "openssl"]
-# naxsi and awscli are not available from the main Alpine repositories yet.
-RUN ["apk", "--no-cache", "--repository=http://dl-cdn.alpinelinux.org/alpine/edge/testing", "add", \
-     "nginx-naxsi=1.16.1-r0", "nginx-naxsi-mod-http-naxsi=1.16.1-r0", "nginx-naxsi-mod-http-xslt-filter=1.16.1-r0"]
+RUN ["apk", "--no-cache", "add", "tini", "dnsmasq", "bash", "curl", "openssl", "python3", "py-pip", "nginx-mod-http-naxsi=1.20.1-r3", "nginx-mod-http-xslt-filter=1.20.1-r3"]
 
-RUN ["apk", "--no-cache", "--repository=http://dl-cdn.alpinelinux.org/alpine/v3.13/community", "add", "aws-cli=1.18.177-r0"]
+RUN ["pip", "install", "awscli~=1.20.0"]
```

Comment:
This is ok, but it's worth checking it's intended that this locks into 1.20 and won't go up to 1.21 or later. The pessimistic locking locks within whatever the most specific version you specified. So `awscli~=1.20.0` means greater than or equal to 1.20.0 and less than 1.21.0, whereas `awscli~=1.20` means anything greater than or equal to 1.20.0 and less than 2.0.0.

This is still fine though but it's an important distinction between the two locks. (I'll still approve this of course since it's all good like this :) 👍